ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryConsistencyFunctionalTest.java

```diff
@@ -331,10 +330,11 @@ public void shouldPassConsistencyTokenInFanOutTableScan()
     // Then:
     assertThat(rows, hasSize(HEADER + 6));
     assertThat(rows.get(HEADER+5).getConsistencyToken(), is(not(Optional.empty())));
-    ConsistencyOffsetVector receivedVector = ConsistencyOffsetVector.deserialize(rows.get(HEADER+5).getConsistencyToken().get().getConsistencyToken());
-    ConsistencyOffsetVector expectedVector = new ConsistencyOffsetVector();
-    expectedVector.setVersion(2);
-    expectedVector.addTopicOffsets("dummy", ImmutableMap.of(5, 5L, 6, 6L, 7, 7L));
+    final ConsistencyOffsetVector receivedVector =
+        ConsistencyOffsetVector.deserialize(rows.get(HEADER+5).getConsistencyToken().get().getConsistencyToken());
+    final ConsistencyOffsetVector expectedVector = new ConsistencyOffsetVector();
+    expectedVector.setVersion(0);
+    expectedVector.addTopicOffsets(topic, ImmutableMap.of(0, 0L, 1, 0L, 2, 2L));
```

Comment:
Since we only update position in the inner store, is that due to the caching effect?

Reply:
I know it is confusing but it is actually correct :) There are 5 records with offsets: partition 0-0, partition 1-0, partition 2-0, partition 2-1, partition 2-2. The offset 0 counts for one record. So, all three partitions have records

Reply:
I see, shall we consider just having one partition in the source and hence in the repartitioned sub-topology then if we do not keep them as 3 partitions with intentional test coverage , since now the result really depends on 1) the aggregate key, and 2) the hashing func we use on repartitioning. Later development not paying attention may accidentally break the test.

Reply:
If I am not mistaken (cc @patrickstuedi), this test needs multiple partitions so that the query gets fanned out to more than one hosts. This way, we test that the consistency vector is forwarded correctly between hosts.